### DISCUSSION TITLE
Config file doesn't work with .ts imports

### DISCUSSION THREAD

**Discussion by himat** [mmkal/slonik-tools#336] [Q&A]
I tried to run `npx slonik-typegen generate --config typegen.config.ts`
but it failed with `Error: Cannot use import statement outside a module`

It seems like I can only use .js files for the config? 

But I need to use a ts import because I want to load my db connection to pass for the `poolConfig` option.

typegen.config.ts
```
import * as dbConn from "./src/dbConnection";

...
```

So is this not possible to do?

**Comment by mmkal** [comment]
You would have to use something like [ts-node](https://npmjs.com/package/ts-node) to make this work. typegen.config.js would still need to be js, but if you add `require('ts-node/register')` at the top you should be able to require typescript files after that.  So your typegen.config.js would look something like this:

```ts
require('ts-node/register')
const dbConn = require('./src/dbConnection')

/** @type {import('@slonik/typegen').Options */
module.exports.default = {
  pool: dbConn.getPool(),
  ...otherOptions
}
```

Edit: if using commonjs you'll probably need to stick to `require` not `import` in typegen.config.js. But you might have some luck by naming it typegen.config.mjs and be able to use `import` that way. I haven't tried that yet.
